core/src/main/java/org/apache/iceberg/rest/responses/ErrorResponseParser.java

```diff
@@ -85,8 +85,9 @@ public static ErrorResponse fromJson(String json) {
   }
 
   public static ErrorResponse fromJson(JsonNode jsonNode) {
-    Preconditions.checkArgument(jsonNode != null && jsonNode.isObject() && jsonNode.has(ERROR),
-        "Cannot parse missing field: error");
+    Preconditions.checkArgument(jsonNode != null && jsonNode.isObject(),
+        "Cannot parse %s from non-object value", ERROR);
```

Comment:
Nit: we would normally include the problem object for debugging.

Reply:
Updated to log the object instead of just `ERROR`. Was a copy-pasta mistake. Good catch!